pkg/oci/routes.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2018 Oracle and/or its affiliates. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/oracle/oci-cloud-controller-manager/pkg/oci/util"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/pkg/errors"
+)
+
+var _ cloudprovider.Routes = &CloudProvider{}
+
+/*
+OCI doesn't have distinct add/remove/update route commands, only the ability to
+update the whole RouteTable.
+
+K8s dispatches up to 200 concurrent creates.
+
+As a result ADD/Create races with itself, especially in light of DELETE.
+
+Ultimately the reconciliation restores order in the galaxy, but only after some
+disruptions in the force.
+
+One _could_ add some locking on the Get/Update of RouteTable, but it's not
+clear that should be required.
+*/
+
+// CreateRoute for OCI, we have to udpate the whole RouteTable not just add the singular route
+func (cp *CloudProvider) CreateRoute(ctx context.Context, clusterName string, nameHint string, route *cloudprovider.Route) error {
+	glog.V(6).Info("Adding route to route table: ", route)
+
+	node, err := cp.NodeLister.Get(string(route.TargetNode))
+
```

Comment:
You're ignoring the error here, so if hte node isn't found in the lister the CCM will panic on the next line.